pymc/model/core.py

```diff
@@ -577,6 +495,17 @@ def __init__(
             functools.partial(str_for_model, formatting="latex"), self
         )
 
+    @classmethod
+    def get_context(
+        cls, error_if_none: bool = True, allow_block_model_access: bool = False
+    ) -> Model | None:
+        model = MODEL_MANAGER.current_context
+        if isinstance(model, BlockModelAccess) and not allow_block_model_access:
+            raise BlockModelAccessError(model.error_msg_on_access)
+        if model is None and error_if_none:
+            raise TypeError("No model on context stack")
+        return model
+
```

Comment:
I'm a bit at a loss here. Why do we even need this class method? Can't it be replaced by a simple call to `modelcontext(None)`?

Reply:
Can you expand on what you are suggesting? 

In main branch, `modelcontext` always returns a model (or raises an error). I can modify `modelcontext` to take `error_if_none: bool = True, allow_block_model_access: bool = False` as arguments and include the logic you highlight above, as well as change the return type from `Model` to `Model | None`. 

Then I'll need to add `@overload`s for those two arguments to maintain correct typing, since there are over 50 locations `modelcontext` is used. I'm happy to do this, but I want to be sure this is what you're intending.

Reply:
These are relevant instances/test where get_context are currently used:

https://github.com/pymc-devs/pymc/blob/938aff4edb0d92fdb0b5d690a1f9d0ebc488e7e7/tests/model/test_core.py#L1212

https://github.com/pymc-devs/pymc/blob/938aff4edb0d92fdb0b5d690a1f9d0ebc488e7e7/pymc/model/core.py#L2091

Reply:
Ok, I see now that the `allow_block_model_access` is what's preventing us from using a `modelcontext` call. I think that it's ok to keep this class method then.